build_tools/circle/build_push_doc.sh

```diff
@@ -32,50 +32,114 @@ git pull origin gh-pages
 git config --global user.email "$GH_EMAIL" > /dev/null 2>&1
 git config --global user.name "$GH_NAME" > /dev/null 2>&1
 
-# remove all files that are not in the .git dir
-find . -not -name ".git/*" -type f -maxdepth 1 -delete
-
-# Remove the remaining directories. Some of these are artifacts of the LAST
-# gh-pages build, and others are remnants of the package itself
-declare -a leftover=(".cache/"
-                     ".idea/"
-                     "build/"
-                     "build_tools/"
-                     "doc/"
-                     "examples/"
-                     "pmdarima/"
-                     "pmdarima.egg-info/"
-                     "_downloads/"
-                     "_images/"
-                     "_modules/"
-                     "_sources/"
-                     "_static/"
-                     "auto_examples/"
-                     "includes"
-                     "modules/")
+
+function deploy() {
+  git add --all
+  git commit -m "[ci skip] Publishing updated documentation for '${CIRCLE_BRANCH}' branch"
+
+  # We have to re-add the origin with the GH_TOKEN credentials
+  git remote rm origin
+  git remote add origin https://${GH_NAME}:${GH_TOKEN}@github.com/${GH_NAME}/pmdarima.git
+
+  # NOW we should be able to push it
+  git push origin gh-pages
+}
+
+
+# If we're on master or develop, we'll end up deploying
+echo "Branch name: ${CIRCLE_BRANCH}"
+
+# Show the present files:
+ls -la
+
+# On both of these, we'll need to remove the artifacts from the package
+# build itself
+declare -a leftover=("benchmarks"
+                     "build"
+                     "dist"
+                     "doc"
+                     "pmdarima"
+                     "pmdarima.egg-info")
 
 # check for each left over file/dir and remove it
 for left in "${leftover[@]}"
 do
-    rm -r ${left} || echo "${left} does not exist; will not remove"
+  echo "Removing ${left}"
+  rm -rf ${left}
 done
 
-# we need this empty file for git not to try to build a jekyll project
-touch .nojekyll
-mv html/* ./
-rm -r html/
+# If it's develop, we can simply rename the "html" directory as the
+# "develop" directory
+if [[ ${CIRCLE_BRANCH} == "develop" ]]; then
+
+  # Remove the existing 'dev' folder (if it's there. might not be the
+  # first time we do this)
+  if [[ -d "develop" ]]; then
+    rm -rf develop/
+  fi
+
+  # Rename the html dir
+  mv html develop
+  # That's it for dev
 
-# Add everything, get ready for commit. But only do it if we're on master
-if [[ "${CIRCLE_BRANCH}" =~ ^master$|^[0-9]+\.[0-9]+\.X$ ]]; then
-    git add --all
-    git commit -m "[ci skip] publishing updated documentation..."
+# Otherwise it's master or another branch, which is a bit more involved.
+# Remove the artifacts from the previous deployment, move the new ones into the
+# folder as well as into the versioned folder. This won't be deployed unless
+# it's master
+else
```

Comment:
Maybe this is naïve, but do we want to go through this whole process if it is not `master` or `develop`? I know we don't deploy, but we could theoretically catch this here and not build docs for feature branches at all.

Reply:
Yes, and actually originally I did exactly that (ignored for other branches), but it occurred to me we want to make sure new PRs will still pass on all this logic before we let them into `develop` or `master` (particularly if we forget to bump a version or anything)

Reply:
Okay, that makes sense!